ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/AssertTopicExecutor.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.execution;
+
+import static io.confluent.ksql.rest.Errors.assertionFailedError;
+import static io.confluent.ksql.util.KsqlConfig.KSQL_ASSERT_TOPIC_DEFAULT_TIMEOUT_MS;
+
+import io.confluent.ksql.KsqlExecutionContext;
+import io.confluent.ksql.execution.expression.tree.Literal;
+import io.confluent.ksql.parser.tree.AssertTopic;
+import io.confluent.ksql.rest.SessionProperties;
+import io.confluent.ksql.rest.entity.AssertTopicEntity;
+import io.confluent.ksql.rest.server.resources.KsqlRestException;
+import io.confluent.ksql.services.KafkaTopicClient;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.RetryUtil;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public final class AssertTopicExecutor {
+
+  private static final int RETRY_MS = 100;
+
+  private AssertTopicExecutor() {
+
+  }
+
+  public static StatementExecutorResponse execute(
+      final ConfiguredStatement<AssertTopic> statement,
+      final SessionProperties sessionProperties,
+      final KsqlExecutionContext executionContext,
+      final ServiceContext serviceContext
+  ) {
+    final KafkaTopicClient client = serviceContext.getTopicClient();
+    final int timeout = statement.getStatement().getTimeout().isPresent()
+        ? (int) statement.getStatement().getTimeout().get().toDuration().toMillis()
+        : executionContext.getKsqlConfig().getInt(KSQL_ASSERT_TOPIC_DEFAULT_TIMEOUT_MS);
+    try {
+      RetryUtil.retryWithBackoff(
+          timeout / RETRY_MS,
+          RETRY_MS,
+          RETRY_MS,
```

Comment:
I *think* this should be timeout.  If I'm reading code correctly, there'd only be one retry with the code as-is.  (I'm skimming, so happy to be wrong.)

Reply:
```
public static void retryWithBackoff(
      final int maxRetries,
      final int initialWaitMs,
      final int maxWaitMs,
      final Runnable runnable,
      final Class<?>... passThroughExceptions)
```

The `retryWithBackoff` method increases the retry timeout between each retry.  It starts at `initialWaitMs` and then inceases to `2*initialWaitMs` and then `4*initialWaitMs` etc until it reaches `maxWait`. So if we keep `initialWaitMs` == `maxWaitMs` then it will retry  every `initialWaitMs` ms until we reach `maxRetries`.